[dynamo] Guard serialization for TYPE_MATCH

zhxchen17 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #152332
* #152331
* #152330
* #152329
* #152328
* #152327
* #152326
* __->__ #152325

Adding guard serialization for TYPE_MATCH

Differential Revision: [D73780438](https://our.internmc.facebook.com/intern/diff/D73780438/)

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @chenyang78 @kadeng @chauhang @amjames

zhxchen17 committed:
[dynamo] Guard serialization for TYPE_MATCH

Adding guard serialization for TYPE_MATCH

Differential Revision: [D73780438](https://our.internmc.facebook.com/intern/diff/D73780438/)

[ghstack-poisoned]

Skylion007 commented on torch/_dynamo/guards.py:
```diff
@@ -524,6 +524,14 @@ def get_key_index_source(source, index):
 
 
+def raise_local_type_error(obj):
+    raise RuntimeError(
```
```suggestion
    raise TypeError(

Skylion007 commented on torch/_dynamo/guards.py:
```diff
@@ -524,6 +524,14 @@ def get_key_index_source(source, index):
     return f"list(dict.keys({source}))[{index}]"
 
 
+def raise_local_type_error(obj):
```
```suggestion
def raise_local_type_error(obj : Any) -> typing.NoReturn:

zhxchen17 committed:
Update on "[dynamo] Guard serialization for TYPE_MATCH"


Adding guard serialization for TYPE_MATCH

Differential Revision: [D73780438](https://our.internmc.facebook.com/intern/diff/D73780438/)

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng chauhang amjames

[ghstack-poisoned]

zhxchen17 committed:
Update on "[dynamo] Guard serialization for TYPE_MATCH"


Adding guard serialization for TYPE_MATCH

Differential Revision: [D73780438](https://our.internmc.facebook.com/intern/diff/D73780438/)

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng chauhang amjames

[ghstack-poisoned]
